independent-projects/arc/runtime/src/main/java/io/quarkus/arc/ArcContainerImpl.java

```diff
@@ -269,6 +269,8 @@ synchronized void shutdown() {
                 ArcCDI arcCdi = (ArcCDI) cdi;
                 arcCdi.destroy();
             }
+            // Terminate request context if for any reason is still active
+            requestContext.terminate();
```

Comment:
Shouldn't we have a test that asserts the order in which contexts shutdown? Singleton/app scoped doesn't matter but it is important that request scope precedes them.

Reply:
Well, I'd preferer if no one (including our tests) relies on this behavior. It's not well defined and could change any time.

Reply:
I wouldn't want to rely on ordering of app and singleton scopes but it makes a lot of sense that 'wider' scopes are destroyed later than those with 'narrower' scope. E.g. req. scope should be destroyed before either app or singleton.